syntax = "proto3";

package image;

service Image {
  rpc UploadImg(FileHeader) returns (UploadFileResult) {}
  rpc DeleteFile(DeleteFileInput) returns (Message) {}
  rpc BulkUpload(MultipleFileHeader) returns (MultipleUploadFileResult) {}
  rpc BulkDeleteFile(MultipleFileIdInput) returns (Message) {}
}

message FileHeader {
  string filename = 1;
  string folder = 2;
  string content = 3;
}

message UploadFileResult {
  string file_id = 1;
  string name = 2;
  string url = 3;
  string content_type = 4;
}

message DeleteFileInput {
  string file_id = 1;
}

message MultipleFileIdInput {
  repeated string file_ids = 1;
}

message Message {
  string message = 1;
}

message FileHeaderWithoutFolder {
  string filename = 1;
  string content = 2;
}

message MultipleFileHeader {
  repeated FileHeaderWithoutFolder files = 1;
  string folder = 2;
}

message MultipleUploadFileResult {
  repeated UploadFileResult datas = 1;
}